### PR TITLE
drivers/opt3001 : Expose Configurations to Kconfig 

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in OPT3001 Sensor Device driver to Kconfig.

### Testing procedure

New macro was introduced in tests/driver_opt3001/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-185-g2e649-Kconfig_opt3001)
CONFIG_OPT3001_I2C_ADDRESS=(0x45)
CONFIG_OPT3001_CONVERSION_TIME=(0x0800)
OPT3001 test application

Initializing OPT3001 sensor at I2C_0 ... opt3001_init: Error reading device ID!
init device [ERROR]

#### Usage with CFLAGS 

/tests/driver_opt3001/Makefile

> CFLAGS += -DCONFIG_OPT3001_I2C_ADDRESS=0x44
> CFLAGS += -DCONFIG_OPT3001_CONVERSION_TIME=OPT3001_CONVERSION_TIME_100_MS

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-185-g2e649-Kconfig_opt3001)
CONFIG_OPT3001_I2C_ADDRESS=(0x44)
CONFIG_OPT3001_CONVERSION_TIME=(0x0000)
OPT3001 test application

Initializing OPT3001 sensor at I2C_0 ... opt3001_init: Error reading device ID!
init device [ERROR]

#### Usage with Kconfig

/tests/driver_opt3001/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-185-g2e649-Kconfig_opt3001)
CONFIG_OPT3001_I2C_ADDRESS=0x47
CONFIG_OPT3001_CONVERSION_TIME=(0x0000)
OPT3001 test application

Initializing OPT3001 sensor at I2C_0 ... opt3001_init: Error reading device ID!
init device [ERROR]

Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri

### CODE REVIEW

File: drivers/opt3001/Kconfig [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+choice
+    bool "OPT3001 conversion time[ms]"
+    default OPT3001_CONVERSION_TIME_800
+    help
+        OPT3001 is capable of integrating results over either 100 ms or 800 ms
+        time periods. Default value is set to 800 ms
+
+config OPT3001_CONVERSION_TIME_100
+    bool "100ms"
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:32]:
```suggestion
    bool "100 ms"
```

File: drivers/opt3001/Kconfig [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+config OPT3001_CONVERSION_TIME_800
+    bool "800ms"
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:35]:
```suggestion
    bool "800 ms"
```

File: drivers/opt3001/Kconfig [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    hex "Default I2C address"
+    range 0x44 0x47
+    default 0x45
+    help
+        OPT3001 allows for upto 4 devices on Single Bus. The value depends on
+        the state of ADDR Pin. Default value (0x45) corresponds to ADDR pin tied
+        to VDD. For more information refer I2C Address Selection (7.3.4.1) in Datasheet.
+
+choice
+    bool "OPT3001 conversion time[ms]"
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:25]:
```suggestion
    bool "OPT3001 conversion time"
```

File: drivers/opt3001/Kconfig [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    default OPT3001_CONVERSION_TIME_800
+    help
+        OPT3001 is capable of integrating results over either 100 ms or 800 ms
+        time periods. Default value is set to 800 ms
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:29]:
```suggestion
        time periods.
```

**Comment by akshaim** [comment]
> Looks good. Just minor doc comments

Thanks for the comments

File: drivers/opt3001/Kconfig [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+if KCONFIG_MODULE_OPT3001
+
+config OPT3001_I2C_ADDRESS
+    hex "Default I2C address"
+    range 0x44 0x47
+    default 0x45
+    help
+        OPT3001 allows for upto 4 devices on Single Bus. The value depends on
+        the state of ADDR Pin. Default value (0x45) corresponds to ADDR pin tied
+        to VDD. For more information refer I2C Address Selection (7.3.4.1) in Datasheet.
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:22]:
```suggestion
        The OPT3001 allows up to 4 devices on single bus. The address depends on
        the state of ADDR Pin. The default value (0x45) corresponds to ADDR pin
        tied to VDD. For more information check the I2C Address Selection
        (7.3.4.1) section in the Datasheet.
```

File: drivers/opt3001/Kconfig [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+choice
+    bool "OPT3001 conversion time"
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:25]:
```suggestion
    bool "Conversion time"
```

File: drivers/opt3001/Kconfig [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    default OPT3001_CONVERSION_TIME_800
+    help
+        OPT3001 is capable of integrating results over either 100 ms or 800 ms
```
Comment (leandrolanzieri) [inline drivers/opt3001/Kconfig:28]:
```suggestion
        Time over which the results are integrated. A longer integration time
        allows for a lower noise measurement.
```

**Comment by leandrolanzieri** [comment]
@akshaim please squash

**Comment by akshaim** [comment]
> @akshaim please squash

Done. Thanks for the comments

**Pull request merged**